frontend/src/concepts/pipelines/content/compareRuns/metricsSection/roc/RocCurveCompare.tsx

```diff
@@ -0,0 +1,125 @@
+import * as React from 'react';
+import {
+  Bullseye,
+  EmptyState,
+  EmptyStateBody,
+  EmptyStateHeader,
+  EmptyStateVariant,
+  Flex,
+  Spinner,
+  Split,
+  SplitItem,
+  Text,
+} from '@patternfly/react-core';
+import DashboardHelpTooltip from '~/concepts/dashboard/DashboardHelpTooltip';
+import { useCheckboxTableBase } from '~/components/table';
+import ROCCurve from '~/concepts/pipelines/content/artifacts/charts/ROCCurve';
+import { RunArtifact } from '~/concepts/pipelines/apiHooks/mlmd/types';
+import {
+  getFullArtifactPathLabel,
+  getFullArtifactPaths,
+  getLinkedArtifactId,
+} from '~/concepts/pipelines/content/compareRuns/metricsSection/utils';
+import { CompareRunsEmptyState } from '~/concepts/pipelines/content/compareRuns/CompareRunsEmptyState';
+import RocCurveTable from './RocCurveTable';
+import { FullArtifactPathsAndConfig } from './types';
+import { isConfidenceMetric, buildRocCurveConfig } from './utils';
+
+type RocCurveCompareProps = {
+  runArtifacts?: RunArtifact[];
+  isLoaded: boolean;
+};
+
+const RocCurveCompare: React.FC<RocCurveCompareProps> = ({ runArtifacts, isLoaded }) => {
+  const [search, setSearch] = React.useState<string>('');
+  const [selected, setSelected] = React.useState<FullArtifactPathsAndConfig[]>([]);
+
+  const configs = React.useMemo(() => {
+    if (!runArtifacts) {
+      return [];
+    }
+
+    const fullArtifactPaths = getFullArtifactPaths(runArtifacts);
+    return fullArtifactPaths
+      .map((fullArtifactPath) => {
+        const customProperties = fullArtifactPath.linkedArtifact.artifact.getCustomPropertiesMap();
+        return {
+          data: customProperties.get('confidenceMetrics')?.getStructValue()?.toJavaScript(),
+          fullArtifactPath,
+        };
+      })
+      .filter((confidenceMetrics) => !!confidenceMetrics.data)
+      .map(({ data, fullArtifactPath }, i) => {
+        // validate the custom properties
+        const confidenceMetricsArray = data?.list as unknown[];
```

Comment:
Why is this case to unknown?

Reply:
becuase it is coming from custom properties

Reply:
i changed the logic to be based on the actual value now